ocaml/otherlibs/systhreads/st_pthreads.h

```diff
@@ -173,8 +173,8 @@ static void st_masterlock_release(st_masterlock * m)
   m->busy = 0;
   // CR ocaml 5 domains: we assume no backup thread
   // st_bt_lock_release(m);
-  custom_condvar_signal(&m->is_free);
   pthread_mutex_unlock(&m->lock);
+  custom_condvar_signal(&m->is_free);
```

Comment:
This change looks great: in a 2-thread case it means that the signal will fire _after_ the owning thread has released the lock, so there's no risk of the woken thread immediately sleeping again to wait for the lock.

Reply:
Indeed, that was my thinking.